File: easy/2996_smallest_missing_integer_greater_than_sequential_prefix_sum/solution.cpp
#include <unordered_set>
#include <vector>

class Solution {
public:
  int missingInteger(std::vector<int> &nums) {
    const size_t n = nums.size();

    int sequentialPrefixSum = nums[0];
    std::unordered_set<int> numsSet = {nums[0]};

    size_t i = 1;
    for (; i < n && nums[i - 1] + 1 == nums[i]; ++i) {
      sequentialPrefixSum += nums[i];
      numsSet.insert(nums[i]);
    }

    for (; i < n; ++i)
      numsSet.insert(nums[i]);

    while (numsSet.find(sequentialPrefixSum) != numsSet.end())
      ++sequentialPrefixSum;

    return sequentialPrefixSum;
  }
};

File: easy/2996_smallest_missing_integer_greater_than_sequential_prefix_sum/solution.py
from typing import List


class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        n = len(nums)

        sequentialPrefixSum = nums[0]
        numsSet = {nums[0]}

        i = 1
        while i < n and nums[i - 1] + 1 == nums[i]:
            sequentialPrefixSum += nums[i]
            numsSet.add(nums[i])
            i += 1

        while i < n:
            numsSet.add(nums[i])
            i += 1

        while sequentialPrefixSum in numsSet:
            sequentialPrefixSum += 1

        return sequentialPrefixSum
